pkg/adaptation/os/ostest/getpid_mock.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright 2021 Andy Dalton
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ostest
+
+// GetpidMock is a component that provides a mock implementation of the
+// os.Getpid() function.  The function returns the configured Pid.
+type GetpidMock struct {
+	Pid int
+}
+
+func (p *GetpidMock) Getpid() int {
+	if p.Pid == 0 {
```

Comment:
Although the API documentation for `os.Getpid()` doesn't explicitly say, I'd think part of its contract would be that it returns a value greater than 0 since a process cannot have a pid with value 0.  Therefore, clients of this API wouldn't expect to handle a value of 0 so returning 0 here would violate the substitution principle.

Reply:
Thanks for the explanation.